dashboard/app/models/peer_review.rb

```diff
@@ -289,6 +289,19 @@ def create_escalated_duplicate
     )
   end
 
+  # Whether this peer review is a review of the latest version of the submitter's answer
+  def current?
+    level_source == submitter.last_attempt(level, script).level_source
```

Comment:
Somewhat less efficient, but a lot easier to reason about as a method on the model.